client/src/main/java/io/pravega/client/segment/impl/SegmentMetadataClient.java

```diff
@@ -18,17 +18,19 @@
     
     /**
      * Returns info for the current segment.
-     * 
+     *
+     * @param delegationToken token to be passed on to segmentstore for validation.
      * @return Metadata about the segment.
      */
-    abstract SegmentInfo getSegmentInfo();
+    abstract SegmentInfo getSegmentInfo(String delegationToken);
```

Comment:
Please don't add credentials to each interface method.   Surely the token acquisition and renewal can be hidden in the `SegmentMetadataClientImpl`?